src/reuse/modules/util/data.ts

```diff
@@ -136,7 +136,7 @@ export class Data {
     for (const key in data) {
       if (typeof data[key] === "object" && !Array.isArray(data[key])) {
         data[key] = this._decryptRecursively(data[key]);
-      } else if (typeof data[key] === "string" || Array.isArray(data[key])) {
+      } else if (typeof data[key] === "string") {
```

Comment:
Basically, the check is correct, since the `global.util.data.decrypt` accepts either a string or an array of strings. Check the function [here](https://github.com/SAP/wdio-qmate-service/blob/a189ae33828c0c7351b7246b1c06f20d9dcd08ca/src/scripts/hooks/utils/decryption.ts#L45)

It would be more correct to add a right type to the global decrypt function [here](https://github.com/SAP/wdio-qmate-service/blob/a189ae33828c0c7351b7246b1c06f20d9dcd08ca/src/reuse/modules/util/Util.ts#L11)

I would suggest that you make it as
```ts
  decrypt: (input: string | Array<string>) => string;
```

Then type check and tests should work fine

Reply:
I think we might also need to update the docs accordingly [here](https://github.com/SAP/wdio-qmate-service/blob/a189ae33828c0c7351b7246b1c06f20d9dcd08ca/src/reuse/modules/util/data.ts#L124)